commit 59de2bebabc5027f93df999d59cc65df591c3e6e
Author: Shaohua Li <shaohua.li@intel.com>
Date:   Mon Apr 20 10:08:35 2009 +1000

    agp: zero pages before sending to userspace
    
    AGP pages might be mapped into userspace finally, so the pages should be
    set to zero before userspace can use it. Otherwise there is potential
    information leakage.
    
    Signed-off-by: Shaohua Li <shaohua.li@intel.com>
    Signed-off-by: Dave Airlie <airlied@redhat.com>

Backported to Debian's 2.6.26 by dann frazier <dannf@debian.org>

diff -urpN linux-source-2.6.26.orig/drivers/char/agp/generic.c linux-source-2.6.26/drivers/char/agp/generic.c
--- linux-source-2.6.26.orig/drivers/char/agp/generic.c	2008-07-13 15:51:29.000000000 -0600
+++ linux-source-2.6.26/drivers/char/agp/generic.c	2009-04-27 21:32:00.000000000 -0600
@@ -1182,7 +1182,7 @@ void *agp_generic_alloc_page(struct agp_
 {
 	struct page * page;
 
-	page = alloc_page(GFP_KERNEL | GFP_DMA32);
+	page = alloc_page(GFP_KERNEL | GFP_DMA32 | __GFP_ZERO);
 	if (page == NULL)
 		return NULL;
 
